uswds/uswds-site
# USWDS-Site - POAM: December ‘24

# Summary

POAM Updates for November + December 2024

## Related issue

https://github.com/uswds/uswds-team/issues/413

Resolves https://github.com/uswds/public-sans/security/dependabot/85

## Preview link

[Preview link:](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cb-poam-nov-24/)

## Major changes

`
rexml updated to `3.4.0`

Before:

```node
6 vulnerabilities (1 low, 4 moderate, 1 high)
```
After:

```node
3 vulnerabilities (0 low, 3 moderate, 0 high)
```

## Testing and review

1. Pull down and run `npm install`
2. Confirm no installation errors
3. Run Compile scripts
4. Confirm no build errors
5. Run `npm run test`
6. Confirm no testing or linting errors 
9. Start local or visit preview link
10. Confirm there are no visual regressions

## Dependency updates

### Gem updates

| **Dependency name** | **Old version** | **New version** |
| --- | --- | --- |
| google-protobuf | 4.28.2-arm64-darwin | 4.29.1-arm64-darwin |
| json | 2.7.2 | 2.9.0 |
| kramdown | 2.4.0 | 2.5.1 | 
| nokogiri | 1.16.7-arm64-darwin |1.17.2-arm64-darwin | 
| rexml | 3.3.8 | 3.4.0 |
| rouge | 4.4.0 | 4.5.1 |
| rspec-core | 3.13.1 | 3.13.2 |
| rspec-support | 3.13.1 | 3.13.2 |
| sass-embedded | 1.79.4-arm64-darwin | 1.83.0-arm64-darwin |
| webrick | 1.8.2 | 1.9.1 |
| zeitwerk | 2.6.18 | 2.7.1 |

### Node updates

| **Dependency name** | **Old version** | **New version** | 
| --- | --- | --- |
| axe-core | ^4.10.0 | ^4.10.2 |
| postcss | ^8.4.47 | ^8.4.49 |
| sass | ^1.79.4 | ^1.83.0 | 
| snyk | ^1.1293.1 | ^1.1294.3 |
